### ISSUE TITLE
GSM vs LTE

### ISSUE THREAD

**Issue by gdt** [mvglasow/satstat#106]
With today's git master, Nexus 5, CM 13, I get

11-10 10:18:03.896 19659-19659/com.vonglasow.michael.satstat W/MainActivity: Got SignalStrength for PHONE_TYPE_GSM but serving cell is not GSM

when on LTE.  This feels like a bug, but I'm not sure.

**Comment by mvglasow** [comment]
This isn't really a bug (and in fact it's been around for a while) but a debug message. I'd implemented it as part of an effort to determine the cell type more accurately. This isn't trivial on Android and is short of black magic, especially when there is no mobile data connection.

Also, Android has three different API calls to report cell information:
* one for the serving cell, unless the serving cell is LTE
* one for neighboring cells, but again, not for LTE cells
* one for all cells, including LTE

However, manufacturers don't always use them in the intended manner: Some support only the first method (and never report any cells other than the serving cell), others violate the spec by reporting LTE cells as GSM/UMTS, others report bogus data. Since there aren't really any apps whose proper functioning depends on that kind of under-the-hood data, there's not much attention to getting these things working properly. (Same story with GPS debug information – I've seen numerous instances of a device not having a fix but claiming to be using 2 out of 2 visible satellites in the fix, which is simply impossible.)

Bottom line: when you see that message, it's either the handset manufacturer reporting bogus data or a sign that SatStat's cell type guessing algorithm needs some tweaking.

**Comment by gdt** [comment]
I wonder if these messages should use some mechanism to count instead of every time after a few times.  In other words, the bug is that it happens, and that's perhaps 1 bug, vs every time.

This is on a Nexus 5 with CM 13, by the way, which feels mainstream but of course mainstream can be buggy.
